lms/djangoapps/grades/models.py

```diff
@@ -363,3 +365,69 @@ def _prepare_params_visible_blocks_id(cls, params):
         """
         params['visible_blocks_id'] = params['visible_blocks'].hash_value
         del params['visible_blocks']
+
+
+class PersistentCourseGrade(TimeStampedModel):
+    """
+    A django model tracking persistent course grades.
+    """
+
+    class Meta(object):
+        # Indices:
+        # (course_id, user_id) for individual grades
+        # (course_id) for instructors to see all course grades
+        # (user_id) for course dashboard
+        unique_together = [
+            ('course_id', 'user_id'),
+        ]
+
+    # primary key will need to be large for this table
+    id = UnsignedBigIntAutoField(primary_key=True)  # pylint: disable=invalid-name
+
+    # uniquely identify this particular grade object
+    user_id = models.IntegerField(blank=False)
```

Comment:
This needs an explicit index on it.  

Indices are ordered, so a (course, user) index can be thought of as kind of like a view of the table,sort by course_id, and in the case of ties, by user_id.  You could use that index to quickly find all the entries for a given course, or for a given course and user, but trying to find all the entries for a given user in any course would kind of be like trying to find all the words that have "e" as the second letter in the dictionary.  You still have to look at everything.

So given a unique_together value of (course_id, user_id), you don't need a separate index on course_id, but you do need one on user_id.